com.ktechpit.ultimate-media-downloader.yaml

```diff
@@ -0,0 +1,80 @@
+app-id: com.ktechpit.ultimate-media-downloader
+runtime: org.kde.Platform
+runtime-version: 5.15-21.08
+sdk: org.kde.Sdk
+command: ultimate-media-downloader
+finish-args:
+  - --share=network
+  - --share=ipc
+  - --socket=fallback-x11
+  - --socket=wayland
+  - --filesystem=xdg-download
+  - --device=dri
+  - --socket=pulseaudio
+  
+tags:
+  - proprietary
+  
+modules:
+  - name: lzo
+    config-opts:
+      - '--enable-shared'
+      - '--disable-static'
+    cleanup:
+      - /include
+      - /share/doc
+      - '*.la'
+      - /lib/*.so
+    sources:
+      - type: archive
+        url: http://www.oberhumer.com/opensource/lzo/download/lzo-2.10.tar.gz
+        sha256: c0f892943208266f9b6543b3ae308fab6284c5c90e627931446fb49b4221a072
+
+  - name: squashfs-tools
+    buildsystem: simple
+    build-commands:
+      - XZ_SUPPORT=1 LZO_SUPPORT=1 make -C squashfs-tools -j ${FLATPAK_BUILDER_N_JOBS} install INSTALL_DIR=${FLATPAK_DEST}/bin
+    sources:
+    - type: git
+      url: https://github.com/plougher/squashfs-tools.git
+      tag: '4.5'
+      commit: '0496d7c3de3e09da37ba492081c86159806ebb07'
+      
+  - name: ultimate-media-downloader
+    buildsystem: simple
+    build-commands:
+      - install process_snap /
+      - install -Dm644 com.ktechpit.ultimate-media-downloader.metainfo.xml /share/metainfo/com.ktechpit.ultimate-media-downloader.metainfo.xml
+    sources:
+      - type: file
+        path: com.ktechpit.ultimate-media-downloader.metainfo.xml
+      - type: file
+        path: ultimate-media-downloader
+      - type: script
+        dest-filename: process_snap
+        commands:
+          - unsquashfs -no-progress ultimate-media-downloader.snap
+          - rm -f ultimate-media-downloader.snap
+          - mkdir bin lib
+          - mkdir -p share/metainfo
+          - mkdir -p share/application
+          - mkdir -p share/icons/hicolor/512x512/apps/
+          - mv squashfs-root/usr/bin/ultimate-media-downloader bin/ultimate-media-downloader
+          - mv squashfs-root/usr/lib/* lib/
+          - mv squashfs-root/meta/gui/icon.png share/icons/hicolor/512x512/apps/com.ktechpit.ultimate-media-downloader.png
+          - mv squashfs-root/usr/share/applications/ultimate-media-downloader.desktop share/application/com.ktechpit.ultimate-media-downloader.desktop
+          - sed -i 's|Icon=.*|Icon=com.ktechpit.ultimate-media-downloader|g' share/application/com.ktechpit.ultimate-media-downloader.desktop
+          - rm -r squashfs-root
+      - type: extra-data
```

Comment:
is the use of `extra-data` really necessary? This is used for things that are not redistributable.  You could just simply unpack the snap in the build process instead, short of using a more appropriate format)

Reply:
removed extras data. Thanks for suggestion :)